dsautils/dsa_store.py

```diff
@@ -277,21 +342,29 @@ def a(event):
                 raise
         return a
 
-    def _parse_value(self, value: "Json String") -> "Dictionary":
+    def _parse_value(self, value: "Json String",
+                     parse_func: "function" = 'default') -> "Dictionary":
         """Parse the string in JSON format into a dictionary.
 
         :param value: JSON string of the form: {"key":"value"}
                       or {"key":number|bool}
+        "param parse_func: Set to None to allow NaN, Infinity and -Infinity
         :type value: String
+        :type key: String (Ex. '/mont/snap/1')
+        :type parse_func: Function which takes a string.
         :return: Key,value dictionary
         :rtype: Dictionary
         :raise: ValueError
         """
 
         self.log.function('_parse_value')
+
+        if parse_func == 'default':
+            parse_func = self._strict_json
+
```

Comment:
duplicate code. replace with function.

Reply:
I don't think I made changes here.

Reply:
right. I'm reviewing my own changes which isn't really appropriate but I tried to be unbiased as possible. I'm making those changes and adding test cases.